diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 3478f10e3..2ea727607 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -28,12 +28,12 @@ file(MAKE_DIRECTORY ${LIB_AUTOGEN_DIR})
 include_directories(${LIB_AUTOGEN_DIR})
 
 if (INTEGRATE_VCPKG)
-    add_library(PROTO_OBJECTS OBJECT "${CMAKE_SOURCE_DIR}/proto/PulsarApi.proto")
+    add_library(PROTO_OBJECTS OBJECT "${PROJECT_SOURCE_DIR}/proto/PulsarApi.proto")
     target_link_libraries(PROTO_OBJECTS PUBLIC protobuf::libprotobuf)
     target_include_directories(PROTO_OBJECTS PUBLIC ${LIB_AUTOGEN_DIR})
     protobuf_generate(
         TARGET PROTO_OBJECTS
-        IMPORT_DIRS ${CMAKE_SOURCE_DIR}/proto
+        IMPORT_DIRS ${PROJECT_SOURCE_DIR}/proto
         PROTOC_OUT_DIR ${LIB_AUTOGEN_DIR})
     set(COMMON_LIBS ${COMMON_LIBS} PROTO_OBJECTS)
 else ()
@@ -73,9 +73,9 @@ if (INTEGRATE_VCPKG)
     target_link_libraries(PULSAR_OBJECT_LIB PROTO_OBJECTS)
 endif ()
 target_include_directories(PULSAR_OBJECT_LIB PUBLIC
-    "${CMAKE_SOURCE_DIR}"
-    "${CMAKE_SOURCE_DIR}/include"
-    "${CMAKE_BINARY_DIR}/include")
+    "${PROJECT_SOURCE_DIR}"
+    "${PROJECT_SOURCE_DIR}/include"
+    "${PROJECT_BINARY_DIR}/include")
 
 include(CheckCXXSymbolExists)
 if (BUILD_DYNAMIC_LIB)
@@ -84,8 +84,8 @@ if (BUILD_DYNAMIC_LIB)
     set_property(TARGET pulsarShared PROPERTY VERSION ${LIBRARY_VERSION})
     target_link_libraries(pulsarShared PRIVATE ${COMMON_LIBS} ${CMAKE_DL_LIBS})
     target_include_directories(pulsarShared PUBLIC
-        ${CMAKE_SOURCE_DIR}
-        ${CMAKE_SOURCE_DIR}/include)
+        ${PROJECT_SOURCE_DIR}
+        ${PROJECT_SOURCE_DIR}/include)
     target_link_directories(pulsarShared PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
     if (MSVC)
         target_include_directories(pulsarShared PRIVATE ${dlfcn-win32_INCLUDE_DIRS})
@@ -104,9 +104,9 @@ endif()
 if (BUILD_STATIC_LIB)
     add_library(pulsarStatic STATIC $<TARGET_OBJECTS:PULSAR_OBJECT_LIB>)
     target_include_directories(pulsarStatic PUBLIC
-        ${CMAKE_BINARY_DIR}/include
-        ${CMAKE_SOURCE_DIR}
-        ${CMAKE_SOURCE_DIR}/include)
+        ${PROJECT_BINARY_DIR}/include
+        ${PROJECT_SOURCE_DIR}
+        ${PROJECT_SOURCE_DIR}/include)
     if (MSVC)
         set_property(TARGET pulsarStatic PROPERTY OUTPUT_NAME "${LIB_NAME}-static")
         target_include_directories(pulsarStatic PRIVATE ${dlfcn-win32_INCLUDE_DIRS})
@@ -206,7 +206,7 @@ if (LINK_STATIC AND BUILD_STATIC_LIB)
             endif()
         endforeach()
 
-        set(PULSAR_WITH_DEPS ${CMAKE_BINARY_DIR}/lib/libpulsarwithdeps.a)
+        set(PULSAR_WITH_DEPS ${PROJECT_BINARY_DIR}/lib/libpulsarwithdeps.a)
         add_custom_target(pulsarStaticWithDeps
                 ALL
                 BYPRODUCTS merged-library
